force-app/main/default/classes/Application_TEST.cls

```diff
@@ -0,0 +1,151 @@
+            insertResult.getErrors()[0].getMessage(),
+            'Should fail with the Open/Due Date validation error message'
+        );
+    }
+
+    /********************************************************
+     * @description Should fail to update an application when setting the due date prior to the open date
+     *********************************************************/
+    @isTest
+    static void shouldFailUpdateWhenApplicationDueDateIsBeforeOpenDate() {
+        Account academicAccount = getAcademicAccount();
+        Contact student = getStudent();
+
+        Application__c openEndedApplication = new Application__c(
+            Applicant__c = student.Id,
+            Applying_To__c = academicAccount.Id,
+            Open_Date__c = System.today()
+        );
+        insert openEndedApplication;
+
+        Test.startTest();
+        openEndedApplication.Due_Date__c = System.today().addDays(-10);
+        Database.SaveResult updateResult = Database.update(openEndedApplication, false);
+        Test.stopTest();
+
+        System.assertEquals(false, updateResult.isSuccess(), 'Update should have failed due to invalid date range');
+        System.assertEquals(
+            Label.stgErrorAppDueDate,
+            updateResult.getErrors()[0].getMessage(),
+            'Should fail with the Open/Due Date validation error message'
+        );
+    }
+
+    /********************************************************
+     * @description Should succeed to insert an application with just one date set
+     *********************************************************/
+    @isTest
+    static void shouldSucceedWhenApplicationHasEitherDueOrOpenDate() {
+        Account academicAccount = getAcademicAccount();
+        Contact student = getStudent();
+
+        Application__c pastDueApplication = new Application__c(
+            Applicant__c = student.Id,
+            Applying_To__c = academicAccount.Id,
+            Due_Date__c = System.today().addDays(-10)
+        );
+        Application__c openEndedApplication = new Application__c(
+            Applicant__c = student.Id,
+            Applying_To__c = academicAccount.Id,
+            Open_Date__c = System.today().addDays(-10)
+        );
+
+        Test.startTest();
+        Database.SaveResult[] insertResults = Database.insert(
+            new List<Application__c>{ pastDueApplication, openEndedApplication },
+            false
+        );
+        Test.stopTest();
+
+        for (Database.SaveResult insertResult : insertResults) {
+            System.assertEquals(
+                true,
+                insertResult.isSuccess(),
+                'Insert should succeed since date fields are not compared due to only one of them being set'
+            );
+        }
+    }
+
+    private static Account getAcademicAccount() {
+        return [SELECT Id FROM Account LIMIT 1];
+    }
+
+    private static Contact getStudent() {
+        return [SELECT Id FROM Contact LIMIT 1];
+    }
```

Comment:
My experience with other tests is that these queries are not sufficiently selective and should use the Id of the Account or Contact generated. This can cause an issue when tests are run in parallel due to the joys of platform quirks.

I would prefer that we be selective in these cases. A different approach would be to insert the Contact and Account in these methods and return them rather than use the test setup. In those cases we are guaranteed to be working with the same records.

This is not a required change.